### ISSUE TITLE
Paste not working in IE11

### ISSUE THREAD

**Issue by einfallstoll** [froala/wysiwyg-editor#550]
I recently discovered a bug with the paste functionality in IE11.
It seems like `DragEvent.clipboardData` is undefined and breaks the paste functionality in IE 11. After some testing and try and error I figured out, that it'll work with `window.clipboardData` instead. So I did a little hack to get it working:

```
if (b && b.clipboardData && b.clipboardData.getData) {
    /* ... */
} /* START FIX */ else if (window.clipboardData) {
    this.clipboardHTML = window.clipboardData.getData('Text');
    return this.processPaste(), b.preventDefault && (b.stopPropagation(), b.preventDefault()), !1
} /* END FIX */
```

**Comment by stefanneculai** [comment]
There is no issue with pasting on the master version in IE11.

**Comment by einfallstoll** [comment]
So the next update will fix this issue?

**Comment by stefanneculai** [comment]
We cannot reproduce the issue on the master version. If you could try the master version and let us know if you still have the problem that would be great.

**Comment by einfallstoll** [comment]
Reproduced on the master version in IE11 on Windows 7 against jQuery 1.11.3.
As mentioned above `b.clipboardData` is not defined.

Here's what I'm debugging:

![Developer Tools in IE11](http://i.imgur.com/2kXYGR6.png)

I guess we can reopen this issue now... Let me know if should test something for you. I could also fix the issue myself and create a Pull Request, but I don't have access to your development repository.

**Comment by stefanneculai** [comment]
`b.clipboardData` might be undefined, but pasting should work anyway. Could you give more details about where you're copying from?

**Comment by einfallstoll** [comment]
It doesn't depend on where I'm copying from. So I tried both some plain text from a text editor as well as predefined content from the editor itself.

The only other behaviour I could describe is that it'll paste `<br></br>` when hitting Ctrl+V or manually rightclick and paste. So instead of the clipboard content it'll insert a line break.

**Comment by stefanneculai** [comment]
We have just tried it again, but we cannot replicate the issue. It is odd that the code is passing the if condition when `b.clipboardData` is undefined. Would it be possible to send us a link to a page where you are doing that? If you don't want to make it public, you send it to us via the contact form from our website.

**Comment by stefanneculai** [comment]
Last commit should fix this as well. Could you please test?

**Comment by sunstorymvp** [comment]
I have the same issue when I am trying to add the code like this (copied from Skype)

Chrome Version 42.0.2311.135 (64-bit), mac os

```xml
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android" xmlns:tools="http://schemas.android.com/tools" android:layout_width="match_parent" android:layout_height="match_parent" tools:context=".MainActivity"> <android.support.v7.widget.RecyclerView android:id="@+id/recycle_view" android:layout_width="match_parent" android:layout_height="match_parent" android:scrollbars="vertical"/>

</RelativeLayout>
```
pasted result:

```html
<p><br>
</p>
```

Can't reproduce it on jsfiddle....


And one more thing, when I add code as plain text (from .txt or chrome dev console) it works, BUT editor goes to the endless loop while updating itself.


![](http://savepic.org/7234245.png)

15 seconds js profiler record after pasting code:

![](http://savepic.org/7258842.png)

hope it will help.

**Comment by stefanneculai** [comment]
@sunstorymvp please test with the master version.

**Comment by sunstorymvp** [comment]
I am using https://github.com/froala/meteor-froala/

$ meteor update froala:editor
Your packages are at their latest compatible versions.

is it possible to update that package?

**Comment by stefanneculai** [comment]
We're updating other packages only after the official release.

**Comment by einfallstoll** [comment]
The current master version works now. I assume the fix was made in 196e437794ab74ba4e805a355509af5bb8a1d0d8 (which fixed #555).

**Comment by Icer2000** [comment]
Hi there, 

there are still issues when pasting e.g. from Microsoft Outlook in IE 11. 

Because e.clipboardData is not defined, the froala editor replaces line brakes (\n) with empty string, leading to words gluing together.

We fixed this in ie11 by polyfilling clipboardData on the originalEvent in before.paste. 

```
var ie11ClipboardFix = function (e, editor, originalEvent) {
// polyfill clipboardData event
       if (!originalEvent.clipboardData) {
           originalEvent.clipboardData = {
               types: "text/plain",
               getData: function(type) {
                   return window.clipboardData.getData("Text");
               },
           };
      }
}
$el.on('froalaEditor.paste.before', ie11ClipboardFix);
```
Is there any possibility to fix this in froala itself?
